lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -264,6 +264,12 @@ def get(self, request, *args, **kwargs):
             if allow_public:
                 handouts_html = get_course_info_section(request, request.user, course, 'handouts')
 
+        elif is_masquerading(request.user, course_key):
+            # Ensure that at a minimum masquerading users still see non-user-specific content
+            course_blocks = get_course_outline_block_tree(request, course_key_string, None)
+            handouts_html = get_course_info_section(request, request.user, course, 'handouts')
+            welcome_message_html = get_current_update_for_user(request, course)
```

Comment:
I am surprised you're able to use the regular request/request.user here. Would have thought you'd need some reference back to the original user. Do you know how this happens to work since technically the request.user is an unenrolled user?